src/store/eventing/staged.ts

```diff
@@ -0,0 +1,68 @@
+import eventManager from '../../common/EventManager';
+import { AppState } from '../types';
+import { getCreatorStaged, CreatorItem, isCreatorStagedRegion, isCreatorStagedHighlight } from '../creator';
+import { Event } from '../../@types';
+
+type Status = 'create' | 'update' | 'cancel';
+
+type Type = 'highlight' | 'region';
+
+type StagedChangeEvent = {
+    status: Status;
+    type: Type;
+};
+
+const getStatus = (prevStaged: CreatorItem, nextStaged: CreatorItem): Status | null => {
+    let status: Status | null = null;
+
+    if (prevStaged === null && nextStaged !== null) {
+        status = 'create';
+    }
+
+    if (prevStaged !== null && nextStaged !== null) {
+        status = 'update';
+    }
```

Comment:
Should we handle `if (prevStaged !== null && nextStaged === null)`? Maybe states is `cancel`?

Reply:
I left it off since our pattern of usage for `cancel` is to use `resetCreatorAction`.